### ISSUE TITLE
Chapter 6: Getting error while seeding data for customer

### ISSUE THREAD

**Issue by imran1992** [vercel/next-learn#306]
Error seeding customers: NeonDbError: db error: ERROR: prepared statement "s416330" does not exist

```
$ node -r dotenv/config ./scripts/seed.js
Created "users" table
Seeded 1 users
Created "customers" table
Error seeding customers: NeonDbError: db error: ERROR: prepared statement "s416330" does not exist

Caused by:
    ERROR: prepared statement "s416330" does not exist
    at execute (/Users/everythingapple/Desktop/iNoor/React/Next/nextjs-dashboard/node_modules/@neondatabase/serverless/index.js:1539:48)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Promise.all (index 5)
    at async seedCustomers (/Users/everythingapple/Desktop/iNoor/React/Next/nextjs-dashboard/scripts/seed.js:106:31)
    at async /Users/everythingapple/Desktop/iNoor/React/Next/nextjs-dashboard/scripts/seed.js:165:3 {
  code: '26000',
  sourceError: undefined
}
node:internal/process/promises:288
            triggerUncaughtException(err, true /* fromPromise */);
            ^

NeonDbError: db error: ERROR: prepared statement "s416330" does not exist

Caused by:
    ERROR: prepared statement "s416330" does not exist
    at execute (/Users/everythingapple/Desktop/iNoor/React/Next/nextjs-dashboard/node_modules/@neondatabase/serverless/index.js:1539:48)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Promise.all (index 5)
    at async seedCustomers (/Users/everythingapple/Desktop/iNoor/React/Next/nextjs-dashboard/scripts/seed.js:106:31)
    at async /Users/everythingapple/Desktop/iNoor/React/Next/nextjs-dashboard/scripts/seed.js:165:3 {
  code: '26000',
  sourceError: undefined
}

```
this error is coming somewhere in the following code

```
const { sql } = require('@vercel/postgres');
const {
  invoices,
  customers,
  revenue,
  users,
} = require('../app/lib/placeholder-data.js');
const bcrypt = require('bcrypt');

........
 // Insert data into the "customers" table
    const insertedCustomers = await Promise.all(
      customers.map(
        (customer) => sql`
        INSERT INTO customers (id, name, email, image_url)
        VALUES (${customer.id}, ${customer.name}, ${customer.email}, ${customer.image_url})
        ON CONFLICT (id) DO NOTHING;
      `,
      ),
    );
```

"@vercel/postgres": "^0.5.0"
"bcrypt": "^5.1.1",
"next": "^14.0.0",
Node : 18.18.2

**Comment by Moyrasc** [comment]
I also have the same error but in my case at the time of inserting invoices data.:

Created "users" table
Seeded 1 users
Created "customers" table
Seeded 10 customers
Created "invoices" table
Error seeding invoices: NeonDbError: db error: ERROR: prepared statement "s648255" does not exist

Caused by:
    ERROR: prepared statement "s648255" does not exist
    at execute (C:\Users\Ariad\nextjs-dashboard2\node_modules\@neondatabase\serverless\index.js:1539:48)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Promise.all (index 5)
    at async seedInvoices (C:\Users\Ariad\nextjs-dashboard2\scripts\seed.js:67:30)
    at async C:\Users\Ariad\nextjs-dashboard2\scripts\seed.js:166:3 {
  code: '26000',
  sourceError: undefined
}
node:internal/process/promises:289
            triggerUncaughtException(err, true /* fromPromise */);
            ^

NeonDbError: db error: ERROR: prepared statement "s648255" does not exist

**Comment by YunusShaikh29** [comment]
I am getting the same error for the revenue table!

**Comment by ansulagrawal** [comment]
can you send seed.js file code

**Comment by lucasgustavodiaz** [comment]
> can you send seed.js file code

```
const { sql } = require('@vercel/postgres');
const {
  invoices,
  customers,
  revenue,
  users,
} = require('../app/lib/placeholder-data.js');
const bcrypt = require('bcrypt');

async function seedUsers() {
  try {
    await sql`CREATE EXTENSION IF NOT EXISTS "uuid-ossp"`;
    // Create the "invoices" table if it doesn't exist
    const createTable = await sql`
      CREATE TABLE IF NOT EXISTS users (
        id UUID DEFAULT uuid_generate_v4() PRIMARY KEY,
        name VARCHAR(255) NOT NULL,
        email TEXT NOT NULL UNIQUE,
        password TEXT NOT NULL
      );
    `;

    console.log(`Created "users" table`);

    // Insert data into the "users" table
    const insertedUsers = await Promise.all(
      users.map(async (user) => {
        const hashedPassword = await bcrypt.hash(user.password, 10);
        return sql`
        INSERT INTO users (id, name, email, password)
        VALUES (${user.id}, ${user.name}, ${user.email}, ${hashedPassword})
        ON CONFLICT (id) DO NOTHING;
      `;
      }),
    );

    console.log(`Seeded ${insertedUsers.length} users`);

    return {
      createTable,
      users: insertedUsers,
    };
  } catch (error) {
    console.error('Error seeding users:', error);
    throw error;
  }
}

async function seedInvoices() {
  try {
    await sql`CREATE EXTENSION IF NOT EXISTS "uuid-ossp"`;

    // Create the "invoices" table if it doesn't exist
    const createTable = await sql`
    CREATE TABLE IF NOT EXISTS invoices (
    id UUID DEFAULT uuid_generate_v4() PRIMARY KEY,
    customer_id UUID NOT NULL,
    amount INT NOT NULL,
    status VARCHAR(255) NOT NULL,
    date DATE NOT NULL
  );
`;

    console.log(`Created "invoices" table`);

    // Insert data into the "invoices" table
    const insertedInvoices = await Promise.all(
      invoices.map(
        (invoice) => sql`
        INSERT INTO invoices (customer_id, amount, status, date)
        VALUES (${invoice.customer_id}, ${invoice.amount}, ${invoice.status}, ${invoice.date})
        ON CONFLICT (id) DO NOTHING;
      `,
      ),
    );

    console.log(`Seeded ${insertedInvoices.length} invoices`);

    return {
      createTable,
      invoices: insertedInvoices,
    };
  } catch (error) {
    console.error('Error seeding invoices:', error);
    throw error;
  }
}

async function seedCustomers() {
  try {
    await sql`CREATE EXTENSION IF NOT EXISTS "uuid-ossp"`;

    // Create the "customers" table if it doesn't exist
    const createTable = await sql`
      CREATE TABLE IF NOT EXISTS customers (
        id UUID DEFAULT uuid_generate_v4() PRIMARY KEY,
        name VARCHAR(255) NOT NULL,
        email VARCHAR(255) NOT NULL,
        image_url VARCHAR(255) NOT NULL
      );
    `;

    console.log(`Created "customers" table`);

    // Insert data into the "customers" table
    const insertedCustomers = await Promise.all(
      customers.map(
        (customer) => sql`
        INSERT INTO customers (id, name, email, image_url)
        VALUES (${customer.id}, ${customer.name}, ${customer.email}, ${customer.image_url})
        ON CONFLICT (id) DO NOTHING;
      `,
      ),
    );

    console.log(`Seeded ${insertedCustomers.length} customers`);

    return {
      createTable,
      customers: insertedCustomers,
    };
  } catch (error) {
    console.error('Error seeding customers:', error);
    throw error;
  }
}

async function seedRevenue() {
  try {
    // Create the "revenue" table if it doesn't exist
    const createTable = await sql`
      CREATE TABLE IF NOT EXISTS revenue (
        month VARCHAR(4) NOT NULL UNIQUE,
        revenue INT NOT NULL
      );
    `;

    console.log(`Created "revenue" table`);

    // Insert data into the "revenue" table
    const insertedRevenue = await Promise.all(
      revenue.map(
        (rev) => sql`
        INSERT INTO revenue (month, revenue)
        VALUES (${rev.month}, ${rev.revenue})
        ON CONFLICT (month) DO NOTHING;
      `,
      ),
    );

    console.log(`Seeded ${insertedRevenue.length} revenue`);

    return {
      createTable,
      revenue: insertedRevenue,
    };
  } catch (error) {
    console.error('Error seeding revenue:', error);
    throw error;
  }
}

(async () => {
  await seedUsers();
  await seedCustomers();
  await seedInvoices();
  await seedRevenue();
})();
```

**Comment by gobeila** [comment]
> I am getting the same error for the revenue table!

https://github.com/vercel/next-learn/blob/main/dashboard/starter-example/scripts/seed.js

**Comment by gobeila** [comment]
> can you send seed.js file code

https://github.com/vercel/next-learn/blob/main/dashboard/starter-example/scripts/seed.js

**Comment by ansulagrawal** [comment]
I did but I didn't got any error

**Comment by Any28Flo** [comment]
same error
![image](https://github.com/vercel/next-learn/assets/25911909/4fa4ef11-8082-4862-a653-7bfde79eb37f)

**Comment by ansulagrawal** [comment]
@Any28Flo or @gobeila can you try to install my code https://github.com/ansulagrawal/nextjs-dashboard and do `npm ci` and then run `npm run seed`. As I did it trice by deleting the db and adding back in vercel. I didn't get any error so.

**Comment by gobeila** [comment]
> I did but I didn't get any error.
> 
> My node version is 18.8.0, npm version : 9.8.1.
> 
> ```
> "@vercel/postgres": "^0.5.0",
> "bcrypt": "^5.1.1",
> "next": "^14.0.0",
> ```

I ran the script with the same versions and I still got the error.

output
`➜  nextjs-dashboard git:(main) ✗ npm run seed       

> seed
> node -r dotenv/config ./scripts/seed.js

Created "users" table
Seeded 1 users
Created "customers" table
Seeded 10 customers
Created "invoices" table
Error seeding invoices: NeonDbError: db error: ERROR: prepared statement "s14261" does not exist

Caused by:
    ERROR: prepared statement "s14261" does not exist
    at execute (/Users/alex/Code/monsieur-maison/repos/nextjs-dashboard/node_modules/@neondatabase/serverless/index.js:1539:48)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Promise.all (index 4)
    at async seedInvoices (/Users/alex/Code/monsieur-maison/repos/nextjs-dashboard/scripts/seed.js:67:30)
    at async /Users/alex/Code/monsieur-maison/repos/nextjs-dashboard/scripts/seed.js:166:3 {
  code: '26000',
  sourceError: undefined
}
node:internal/process/promises:288
            triggerUncaughtException(err, true /* fromPromise */);
            ^

NeonDbError: db error: ERROR: prepared statement "s14261" does not exist

Caused by:
    ERROR: prepared statement "s14261" does not exist
    at execute (/Users/alex/Code/monsieur-maison/repos/nextjs-dashboard/node_modules/@neondatabase/serverless/index.js:1539:48)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Promise.all (index 4)
    at async seedInvoices (/Users/alex/Code/monsieur-maison/repos/nextjs-dashboard/scripts/seed.js:67:30)
    at async /Users/alex/Code/monsieur-maison/repos/nextjs-dashboard/scripts/seed.js:166:3 {
  code: '26000',
  sourceError: undefined
}

Node.js v18.18.2`


package.json
`{
  "private": true,
  "scripts": {
    "build": "next build",
    "dev": "next dev",
    "start": "next start",
    "seed": "node -r dotenv/config ./scripts/seed.js"
  },
  "dependencies": {
    "@heroicons/react": "^2.0.18",
    "@tailwindcss/forms": "^0.5.6",
    "@types/node": "20.5.7",
    "@vercel/postgres": "^0.5.0",
    "autoprefixer": "10.4.15",
    "bcrypt": "^5.1.1",
    "clsx": "^2.0.0",
    "next": "^14.0.0",
    "postcss": "8.4.31",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "tailwindcss": "3.3.3",
    "typescript": "5.2.2",
    "zod": "^3.22.2"
  },
  "devDependencies": {
    "@types/bcrypt": "^5.0.1",
    "@types/react": "18.2.21",
    "@types/react-dom": "18.2.14",
    "dotenv": "^16.3.1",
    "prettier": "^3.0.3"
  },
  "engines": {
    "node": ">=18"
  }
}
`

**Comment by Parth86** [comment]
> seed
> node -r dotenv/config ./scripts/seed.js

(node:9215) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 close listeners added to [TLSSocket]. Use emitter.setMaxListeners() to increase limit
(Use `node --trace-warnings ...` to show where the warning was created)
Created "users" table
Seeded 1 users
Created "customers" table
(node:9215) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 close listeners added to [TLSSocket]. Use emitter.setMaxListeners() to increase limit
(node:9215) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 close listeners added to [TLSSocket]. Use emitter.setMaxListeners() to increase limit
(node:9215) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 close listeners added to [TLSSocket]. Use emitter.setMaxListeners() to increase limit
(node:9215) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 close listeners added to [TLSSocket]. Use emitter.setMaxListeners() to increase limit
Seeded 10 customers
Created "invoices" table
Error seeding invoices: NeonDbError: db error: ERROR: prepared statement "s1772643" does not exist

Caused by:
    ERROR: prepared statement "s1772643" does not exist
    at execute (/Users/parthtejpal/Desktop/nextjs-dashboard/node_modules/@neondatabase/serverless/index.js:1539:48)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Promise.all (index 1)
    at async seedInvoices (/Users/parthtejpal/Desktop/nextjs-dashboard/scripts/seed.js:67:30)
    at async /Users/parthtejpal/Desktop/nextjs-dashboard/scripts/seed.js:166:3 {
  code: '26000',
  sourceError: undefined
}
node:internal/process/promises:289
            triggerUncaughtException(err, true /* fromPromise */);
            ^

NeonDbError: db error: ERROR: prepared statement "s1772643" does not exist

Caused by:
    ERROR: prepared statement "s1772643" does not exist
    at execute (/Users/parthtejpal/Desktop/nextjs-dashboard/node_modules/@neondatabase/serverless/index.js:1539:48)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Promise.all (index 1)
    at async seedInvoices (/Users/parthtejpal/Desktop/nextjs-dashboard/scripts/seed.js:67:30)
    at async /Users/parthtejpal/Desktop/nextjs-dashboard/scripts/seed.js:166:3 {
  code: '26000',
  sourceError: undefined
}

Node.js v20.8.0

**Comment by ansulagrawal** [comment]
I tried again now I also got the error.

**Comment by carlosazaustre** [comment]
Same error here.
- Node version: v20.6.1
- NPM version: v9.8.1
- @vercel/postgres version: ^0.5.0

Command:
```sh
$ npm run seed
```

Output
```sh
> seed
> node -r dotenv/config ./scripts/seed.js

Created "users" table
Seeded 1 users
Created "customers" table
Error seeding customers: NeonDbError: db error: ERROR: prepared statement "s1809324" does not exist

Caused by:
    ERROR: prepared statement "s1809324" does not exist
    at execute (/Users/carlosazaustre/WORKSPACE/REPOS/nextjs-dashboard/node_modules/@neondatabase/serverless/index.js:1539:48)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Promise.all (index 5)
    at async seedCustomers (/Users/carlosazaustre/WORKSPACE/REPOS/nextjs-dashboard/scripts/seed.js:106:31)
    at async /Users/carlosazaustre/WORKSPACE/REPOS/nextjs-dashboard/scripts/seed.js:165:3 {
  code: '26000',
  sourceError: undefined
}
node:internal/process/promises:289
            triggerUncaughtException(err, true /* fromPromise */);
            ^

NeonDbError: db error: ERROR: prepared statement "s1809324" does not exist

Caused by:
    ERROR: prepared statement "s1809324" does not exist
    at execute (/Users/carlosazaustre/WORKSPACE/REPOS/nextjs-dashboard/node_modules/@neondatabase/serverless/index.js:1539:48)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Promise.all (index 5)
    at async seedCustomers (/Users/carlosazaustre/WORKSPACE/REPOS/nextjs-dashboard/scripts/seed.js:106:31)
    at async /Users/carlosazaustre/WORKSPACE/REPOS/nextjs-dashboard/scripts/seed.js:165:3 {
  code: '26000',
  sourceError: undefined
}
```

**Comment by gobeila** [comment]
> @Any28Flo or @gobeila can you try to install my code https://github.com/ansulagrawal/nextjs-dashboard and do `npm ci` and then run `npm run seed`. As I did it trice by deleting the db and adding back in vercel. I didn't get any error so.

I still have the same error with your repository.

```
➜  git clone https://github.com/ansulagrawal/nextjs-dashboard.git
Cloning into 'nextjs-dashboard'...
remote: Enumerating objects: 230, done.
remote: Counting objects: 100% (230/230), done.
remote: Compressing objects: 100% (136/136), done.
remote: Total 230 (delta 98), reused 209 (delta 77), pack-reused 0
Receiving objects: 100% (230/230), 998.34 KiB | 9.24 MiB/s, done.
Resolving deltas: 100% (98/98), done.
➜  cd nextjs-dashboard 
➜  nextjs-dashboard git:(master) npm ci             

added 404 packages, and audited 405 packages in 4s

125 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
➜  nextjs-dashboard git:(master) vi .env           
➜  nextjs-dashboard git:(master) npm run seed

> seed
> node -r dotenv/config ./scripts/seed.js

Created "users" table
Seeded 1 users
Created "customers" table
Seeded 10 customers
Created "invoices" table
Error seeding invoices: NeonDbError: db error: ERROR: prepared statement "s15254" does not exist

Caused by:
    ERROR: prepared statement "s15254" does not exist
    at execute (/Users/alex/Code/monsieur-maison/repos/temp/nextjs-dashboard/node_modules/@neondatabase/serverless/index.js:1539:48)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Promise.all (index 3)
    at async seedInvoices (/Users/alex/Code/monsieur-maison/repos/temp/nextjs-dashboard/scripts/seed.js:67:30)
    at async /Users/alex/Code/monsieur-maison/repos/temp/nextjs-dashboard/scripts/seed.js:166:3 {
  code: '26000',
  sourceError: undefined
}
node:internal/process/promises:288
            triggerUncaughtException(err, true /* fromPromise */);
            ^

NeonDbError: db error: ERROR: prepared statement "s15254" does not exist

Caused by:
    ERROR: prepared statement "s15254" does not exist
    at execute (/Users/alex/Code/monsieur-maison/repos/temp/nextjs-dashboard/node_modules/@neondatabase/serverless/index.js:1539:48)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Promise.all (index 3)
    at async seedInvoices (/Users/alex/Code/monsieur-maison/repos/temp/nextjs-dashboard/scripts/seed.js:67:30)
    at async /Users/alex/Code/monsieur-maison/repos/temp/nextjs-dashboard/scripts/seed.js:166:3 {
  code: '26000',
  sourceError: undefined
}

Node.js v18.18.2
```

**Comment by ansulagrawal** [comment]
@gobeila check the table I think user might have been created.

**Comment by delbaoliveira** [comment]
Hey 👋🏼 I was able to reproduce this with a fresh project. Looking into it.

**Comment by gobeila** [comment]
> @gobeila check the table I think user might have been created.

@ansulagrawal as I mentioned earlier in the thread, it partially works but the error seems to occur randomly during the execution. It looks like some weird driver or pooling issue.

**Comment by ansulagrawal** [comment]
yes I also think so , sorry for inconvenience

**Comment by gobeila** [comment]
> yes I also think so , sorry for inconvenience

No worries, thank you for your help.

**Comment by Maniload** [comment]
Maybe this is caused by a multiple `INSERT`s being executed concurrently? It seems to work better when I convert the `Promise.all` calls to simple `for`-loops:

```js
// Insert data into the "invoices" table
const insertedInvoices = [];
for (const invoice of invoices) {
  insertedInvoices.push(await sql`
    INSERT INTO invoices (customer_id, amount, status, date)
    VALUES (${invoice.customer_id}, ${invoice.amount}, ${invoice.status}, ${invoice.date})
    ON CONFLICT (id) DO NOTHING;
  `);
}
```

**Comment by martinmilani** [comment]
> Maybe this is caused by a multiple `INSERT`s being executed concurrently? It seems to work better when I convert the `Promise.all` calls to simple `for`-loops:
> 
> ```js
> // Insert data into the "invoices" table
> const insertedInvoices = [];
> for (const invoice of invoices) {
>   insertedInvoices.push(await sql`
>     INSERT INTO invoices (customer_id, amount, status, date)
>     VALUES (${invoice.customer_id}, ${invoice.amount}, ${invoice.status}, ${invoice.date})
>     ON CONFLICT (id) DO NOTHING;
>   `);
> }
> ```

It works, thanks!

**Comment by carlosazaustre** [comment]
> Hey 👋🏼 I was able to reproduce this with a fresh project. Looking into it.

I created a new start. New seed command throws the same error, but in Vercel Postgres console I have the tables with data.
<img width="1090" alt="image" src="https://github.com/vercel/next-learn/assets/650752/782f61ef-fcf5-4232-9e6b-63a9011a9d3a">

**Comment by Moyrasc** [comment]
> I created a new start. New seed command throws the same error, but in Vercel Postgres console I have the tables with data.

But I don't think it includes all the data, does it?

**Comment by ansulagrawal** [comment]
It doesn't add all values

**Comment by Parth86** [comment]
> Maybe this is caused by a multiple `INSERT`s being executed concurrently? It seems to work better when I convert the `Promise.all` calls to simple `for`-loops:
> 
> ```js
> // Insert data into the "invoices" table
> const insertedInvoices = [];
> for (const invoice of invoices) {
>   insertedInvoices.push(await sql`
>     INSERT INTO invoices (customer_id, amount, status, date)
>     VALUES (${invoice.customer_id}, ${invoice.amount}, ${invoice.status}, ${invoice.date})
>     ON CONFLICT (id) DO NOTHING;
>   `);
> }
> ```

This is working. Thanks

**Comment by ansulagrawal** [comment]
@Maniload thanks for suggesting. Based on your code I have updated the file and added in pr.

Thanks once again.

**Comment by delbaoliveira** [comment]
Here's the fix: https://github.com/vercel/next-learn/pull/314

I believe the problem was with the fact `sql` opens a new db connection whenever it's called. 

@Maniload was on the right track in the sense that using a `for` loop would open the connections sequentially, whereas `Promise.all` initiates the promises in parallel, therefore creating multiple connections at once.

However, with the loop, we're still creating multiple connections. The ideal solution is to run the script on a **single** connection, and we can use `db.client` for that. More details on the PR.

Please let me know if you run into any more issues (and remember to drop your tables on Vercel before testing again) 😄

**Comment by ansulagrawal** [comment]
Yes using pool is best way as its create a single connection.

**Comment by Maniload** [comment]
> Here's the fix: #314
> 
> I believe the problem was with the fact `sql` opens a new db connection whenever it's called.
> 
> @Maniload was on the right track in the sense that using a `for` loop would open the connections sequentially, whereas `Promise.all` initiates the promises in parallel - creating multiple connections at once.
> 
> However, with the loop, we're still creating multiple connections. The ideal solution is to run the script on a **single** connection, and we can use `db.client` for that. More details on the PR.
> 
> Please let me know if you run into any more issues (and remember to drop your tables on Vercel before testing again) 😄

This is definitely the better solution ^^ 

Although now I'm curious. I didn't know `sql` would open a new db connection each time. Isn't this bad for performance when I, for example, do multiple queries while rendering a page?

**Comment by carlosazaustre** [comment]
> Here's the fix: #314
> 
> I believe the problem was with the fact `sql` opens a new db connection whenever it's called.
> 
> @Maniload was on the right track in the sense that using a `for` loop would open the connections sequentially, whereas `Promise.all` initiates the promises in parallel - creating multiple connections at once.
> 
> However, with the loop, we're still creating multiple connections. The ideal solution is to run the script on a **single** connection, and we can use `db.client` for that. More details on the PR.
> 
> Please let me know if you run into any more issues (and remember to drop your tables on Vercel before testing again) 😄

Works perfect! Thank you @delbaoliveira !!
